src/main/java/io/confluent/connect/hdfs/HdfsSinkConnectorConfig.java

```diff
@@ -146,100 +198,223 @@
-      .define(TIMEZONE_CONFIG, Type.STRING, TIMEZONE_DEFAULT, Importance.MEDIUM, TIMEZONE_DOC)
-      .define(FILENAME_OFFSET_ZERO_PAD_WIDTH_CONFIG, Type.INT,
-              FILENAME_OFFSET_ZERO_PAD_WIDTH_DEFAULT, ConfigDef.Range.atLeast(0),
-              Importance.LOW, FILENAME_OFFSET_ZERO_PAD_WIDTH_DOC)
-      .define(HDFS_AUTHENTICATION_KERBEROS_CONFIG, Type.BOOLEAN,
-              HDFS_AUTHENTICATION_KERBEROS_DEFAULT, Importance.HIGH,
-              HDFS_AUTHENTICATION_KERBEROS_DOC)
-      .define(CONNECT_HDFS_PRINCIPAL_CONFIG, Type.STRING, CONNECT_HDFS_PRINCIPAL_DEFAULT,
-              Importance.HIGH, CONNECT_HDFS_PRINCIPAL_DOC)
-      .define(CONNECT_HDFS_KEYTAB_CONFIG, Type.STRING, CONNECT_HDFS_KEYTAB_DEFAULT,
-              Importance.HIGH, CONNECT_HDFS_KEYTAB_DOC)
-      .define(HDFS_NAMENODE_PRINCIPAL_CONFIG, Type.STRING, HDFS_NAMENODE_PRINCIPAL_DEFAULT,
-              Importance.HIGH, HDFS_NAMENODE_PRINCIPAL_DOC)
-      .define(KERBEROS_TICKET_RENEW_PERIOD_MS_CONFIG, Type.LONG,
-              KERBEROS_TICKET_RENEW_PERIOD_MS_DEFAULT, Importance.LOW,
-              KERBEROS_TICKET_RENEW_PERIOD_MS_DOC);
+  private static class SchemaCompatibilityRecommender extends BooleanParentRecommender {
+    
+    public SchemaCompatibilityRecommender() {
+      super(HIVE_INTEGRATION_CONFIG);
+    }
+      
+    @Override
+    public List<Object> validValues(String name, Map<String, String> connectorConfigs) {
+      boolean hiveIntegration = Boolean.parseBoolean(connectorConfigs.get(parentConfigName));
+      if (hiveIntegration) {
+        return Arrays.<Object>asList("BACKWARD", "FORWARD", "FULL");
+      } else {
+        return Arrays.<Object>asList("NONE", "BACKWARD", "FORWARD", "FULL");
+      }
+    }
+
+    @Override
+    public boolean visible(String name, Map<String, String> connectorConfigs) {
+      return true;
+    }
+  }
+  
+  private static class BooleanParentRecommender implements ConfigDef.Recommender {
+    
+    protected String parentConfigName;
+    
+    public BooleanParentRecommender(String parentConfigName) {
+      this.parentConfigName = parentConfigName;
+    }
+    
+    @Override
+    public List<Object> validValues(String name, Map<String, String> connectorConfigs) {
+      return new LinkedList<>();
+    }
+
+    @Override
+    public boolean visible(String name, Map<String, String> connectorConfigs) {
+      return Boolean.parseBoolean(connectorConfigs.get(parentConfigName));
+    }
+  }
+
+  private static class PartitionerClassDependentsRecommender implements ConfigDef.Recommender {
+
+    @Override
+    public List<Object> validValues(String name, Map<String, String> connectorConfigs) {
+      return new LinkedList<>();
+    }
+
+    @Override
+    public boolean visible(String name, Map<String, String> connectorConfigs) {
+      String partitionerName = connectorConfigs.get(PARTITIONER_CLASS_CONFIG);
+      if (partitionerName == null) {
+        partitionerName = DefaultPartitioner.class.getName();
```

Comment:
Hmm, so this seems like a pretty significant drawback -- all the logic around parsing that the config classes can already do is not reusable which means connectors will be duplicating lots of code.

Is there any way we can reuse that parsing support? I'm thinking either an AbstractConfig-like class that allows some fields to fail to parse somehow, or maybe just support that in AbstractConfig itself?